src/plugins/core/figures.ts

```diff
@@ -53,6 +53,44 @@ export class FigurePlugin extends CorePlugin<FigureState> implements FigureState
       case "DELETE_FIGURE":
         this.removeFigure(cmd.id, cmd.sheetId);
         break;
+      case "REMOVE_COLUMNS_ROWS":
+        this.onRowColDelete(cmd.sheetId, cmd.dimension);
+    }
+  }
+
+  private onRowColDelete(sheetId: string, dimension: string) {
+    dimension === "ROW" ? this.onRowDeletion(sheetId) : this.onColDeletion(sheetId);
+  }
+
+  private onRowDeletion(sheetId: string) {
+    const numHeader = this.getters.getNumberRows(sheetId);
+    let gridHeight = 0;
+    for (let i = 0; i < numHeader; i++) {
+      gridHeight += this.getters.getRowSize(sheetId, i);
+    }
+    const figures = this.getters.getFigures(sheetId);
+    for (const figure of figures) {
+      const newFigure = { ...figure };
```

Comment:
since you are only computing `newY` and dispatch it as a partial definition of a figure (line 76), you do not have to destructure it. Destructuring would make sense if you wanted to modify some parts of the object then use that modified object as a whole; but since we can simply dispatch isolated attributed in the [command](https://github.com/odoo/o-spreadsheet/blob/saas-16.2/src/types/commands.ts#L408), it is not necessary. Same goes for the other function.

Reply:
If you are referring to destructuring of figure > newFigure, then yes that indeed seems additional to me now, I have removed it and modified the code slightly. Let me know if this is what you meant. Thanks